operations/pyroscope/helm/pyroscope/templates/deployments-statefulsets.yaml

```
@@ -63,7 +63,7 @@ spec:
             - "-self-profiling.disable-push=true"
             - "-server.http-listen-port={{ $values.service.port }}"
             - "-memberlist.cluster-label={{ .Release.Namespace }}-{{ include "pyroscope.fullname" .}}"
-            - "-memberlist.join=dns+{{ include "pyroscope.fullname" .}}-memberlist.{{ .Release.Namespace }}.svc.cluster.local.:{{ .Values.pyroscope.memberlist.port }}"
+            - "-memberlist.join=dns+{{ include "pyroscope.fullname" .}}-memberlist.{{ .Release.Namespace }}.svc.{{ include "cluster.domain" . }}:{{ .Values.pyroscope.memberlist.port }}"
```

If I understood you correctly you wanted to be able to set things to `pyroscope-memberlist.my-namespace.svc`, we then need to remove the extra dot into the values file:

```suggestion
            - "-memberlist.join=dns+{{ include "pyroscope.fullname" .}}-memberlist.{{ .Release.Namespace }}.svc{{ include "cluster.domain" . }}:{{ .Values.pyroscope.memberlist.port }}"
```